### PR TITLE
[wasm] upgrade emsdk version to 2.0.23

### PR DESCRIPTION
**Pull request by fs-eire**
**Description**: upgrade emsdk to 2.0.23. 
- libcxx updated to llvm-12
- add flag `-Wno-deprecated-copy` for gmock (otherwise it will fail the build for webassembly)
- re-patch file_packager.py

### CODE REVIEW

**Comment by snnn** [comment]
Good luck! If onnx runtime can be built with llvm 12...

**Comment by fs-eire** [comment]
> Good luck! If onnx runtime can be built with llvm 12...
> 
> GPU part doesn't work. CPU-only code probably is fine.

Yes and fortunately we are not going to use CUDA in web assembly anyway.

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -630,6 +630,10 @@ if (onnxruntime_BUILD_UNIT_TESTS)
     set_msvc_c_cpp_compiler_warning_level(3)
     set_target_properties(gmock PROPERTIES FOLDER "External/GTest")
     set_target_properties(gmock_main PROPERTIES FOLDER "External/GTest")
+    if(HAS_DEPRECATED_COPY)
+      target_compile_options(gmock PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:-Wno-deprecated-copy>")
+      target_compile_options(gmock_main PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:-Wno-deprecated-copy>")
```
Comment (snnn) [inline cmake/CMakeLists.txt:635]: Try:
```
target_compile_options(gmock PRIVATE "-w")
target_compile_options(gmock_main PRIVATE "-w")
```

without the if statement of HAS_DEPRECATED_COPY.

File: cmake/CMakeLists.txt [hunk 2]
```diff
@@ -629,6 +629,8 @@ if (onnxruntime_BUILD_UNIT_TESTS)
     add_subdirectory(${PROJECT_SOURCE_DIR}/external/googletest EXCLUDE_FROM_ALL)
     set_msvc_c_cpp_compiler_warning_level(3)
     set_target_properties(gmock PROPERTIES FOLDER "External/GTest")
+    # disable treating all warnings as errors for gmock
+    target_compile_options(gmock PRIVATE "-w")
```
Comment (hanbitmyths) [inline cmake/CMakeLists.txt:633]: Is this intentionally for all builds?

Comment (snnn) [reply cmake/CMakeLists.txt:633]: It's fine.

**Review by snnn — APPROVED** [review]

**Pull request merged**